foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/mappings/AggregateCollectionMapping.java

```diff
@@ -2561,21 +2548,23 @@ protected InsertObjectQuery getInsertObjectQuery(AbstractSession session, ClassD
     public InsertObjectQuery getAndPrepareModifyQueryForInsert(ObjectLevelModifyQuery originalQuery, Object object) {
         AbstractSession session = originalQuery.getSession();
         ClassDescriptor objReferenceDescriptor = getReferenceDescriptor(object.getClass(), session);
-        InsertObjectQuery insertQueryFromDescriptor = getInsertObjectQuery(session, objReferenceDescriptor);
-        insertQueryFromDescriptor.checkPrepare(session, insertQueryFromDescriptor.getModifyRow());
+        InsertObjectQuery insertQuery = getInsertObjectQuery(session, objReferenceDescriptor);
 
-        InsertObjectQuery insertQuery = (InsertObjectQuery)insertQueryFromDescriptor.clone();
         insertQuery.setObject(object);
         insertQuery.setDescriptor(objReferenceDescriptor);
 
-        AbstractRecord targetForeignKeyRow = new DatabaseRecord();
```

Comment:
this part should have stayed as it was - objectAddedDuringUpdate and pre/postInsert callers are doing what is necessary, sorry I didn't check callers of this method earlier

Reply:
Ok, I can add this part back. Interesting that we need to make sure callers of this method are setting the right things. I assume you are referencing:

```java
    ContainerPolicy.copyMapDataToRow(extraData, insertQuery.getModifyRow());
    if(this.listOrderField != null && extraData != null) {
        insertQuery.getModifyRow().put(this.listOrderField, extraData.get(this.listOrderField));
    }
```

Reply:
yes, and slightly different code bellow from other 2 callers from within the same class:
```java
ContainerPolicy.copyMapDataToRow(cp.getKeyMappingDataForWriteQuery(wrappedObject, query.getSession()), insertQuery.getModifyRow());
if(this.listOrderField != null) {
    insertQuery.getModifyRow().add(this.listOrderField, index++);
}
```